### PR TITLE
Added the resitricted_location change for ASR

### PR DESCRIPTION
**Pull request by Rohit25negi**
This is to apply the location restrictions so that a user who is not assigned a location can't access it.
This is similar to the work done here: https://github.com/dimagi/commcare-hq/pull/28290.
This change reuses what is done in the PR above.
So, should not be merged before the above PR gets merged.

### CODE REVIEW

**Comment by mkangia** [comment]
You can use the base branch as the branch of the PR this is waiting for, I think that will set things up for you.

Also, can you explain how adding this method changes the locations accessed?

**Comment by Rohit25negi** [comment]
> You can use the base branch as the branch of the PR this is waiting for, I think that will set things up for you.
> 
> Also, can you explain how adding this method changes the locations accessed?

Hey,
ASR right not using [this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/locations/util.py#L35) to fetch the Location hierarchy which is similar to how MPR pulls its hierarchy(MPR location function is an exact copy of this with test location filter).
Change in this PR will make ASR reuse the MPR location function which will result in location restriction being applied because we are already applying that in other MPR PR linked in the description

File: custom/icds_reports/filters.py [hunk 1]
```diff
@@ -105,6 +105,10 @@ class IcdsRestrictedLocationFilter(AsyncLocationFilter):
     def location_hierarchy_config(self):
         return location_hierarchy_config(self.domain)
 
+    def load_locations_json(self, loc_id):
+        show_test = self.request.GET.get('include_test', False)
+        return load_locs_json(self.domain, loc_id, user=self.request.couch_user, show_test=show_test)
+
```
Comment (mkangia) [inline custom/icds_reports/filters.py:111]: Do we need two separate filters `IcdsLocationFilter` and `IcdsRestrictedLocationFilter` ?
They are now the same with the only difference that `IcdsRestrictedLocationFilter` uses a limited location hierarchy. I guess we could just keep one but I am not sure, just floating the idea. 
Feel free to punt that if that does not look obvious/straight forward/unnecessary

Comment (mkangia) [reply custom/icds_reports/filters.py:111]: `IcdsRestrictedLocationFilter` could just inherit from `IcdsLocationFilter`

**Review by mkangia — APPROVED** [review]

File: custom/icds_reports/filters.py (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/filters.py:111]: thanks for that idea. Let me check

Comment (Rohit25negi) [reply custom/icds_reports/filters.py]: Make it inherit the MPR location filter class

**Review by mkangia — APPROVED** [review]

**Review by calellowitz — APPROVED** [review]

**Pull request closed without merging**